src/plugins/links/public/editor/open_editor_flyout.tsx

```diff
@@ -77,45 +81,35 @@ export async function openEditorFlyout(
       if (!overlayTracker) editorFlyout.close();
     });
 
-    const onSaveToLibrary = async (newLinks: Link[], newLayout: LinksLayoutType) => {
-      const newAttributes = {
-        ...attributes,
+    const onSaveToLibrary = async (newLinks: ResolvedLink[], newLayout: LinksLayoutType) => {
+      const newState: LinksRuntimeState = {
+        ...initialState,
         links: newLinks,
         layout: newLayout,
       };
-      const updatedInput = (initialInput as LinksByReferenceInput).savedObjectId
-        ? await attributeService.wrapAttributes(newAttributes, true, initialInput)
-        : await runSaveToLibrary(newAttributes, initialInput);
-      if (!updatedInput) {
-        return;
-      }
-      resolve({
-        newInput: updatedInput,
 
-        // pass attributes via attributes so that the Dashboard can choose the right panel size.
-        attributes: newAttributes,
-      });
-      parentDashboard?.reload();
+      if (initialState?.savedObjectId) {
+        const { attributes, references } = serializeLinksAttributes(newState);
+        await linksClient.update({
+          id: initialState.savedObjectId,
+          data: attributes,
+          options: { references },
+        });
```

Comment:
Hmm... Because the ID doesn't change, we aren't telling the Dashboard to refetch the saved object. See the following example with two if the same by-reference links panels:


https://github.com/elastic/kibana/assets/8698078/adaa3a61-5940-4882-adda-450d1b19adcd

This is probably out-of-scope of this PR, but interested in thoughts on how big of a deal this is? cc @ThomThomson @nreese